playground/miden-wasm/src/lib.rs

```diff
@@ -142,6 +159,67 @@ pub fn prove_program(asm: &str, inputs_frontend: &str) -> Outputs {
     result
 }
 
+/// Debuggs the program with the given inputs
+#[wasm_bindgen]
+pub fn debug_program(asm: &str, inputs_frontend: &str, steps: usize) -> DebugOutputs {
```

Comment:
This seems quite inefficient, for example, lets say we are on step 10_000, to get into step 10_001 it seems the program has to be instantiated again and re-executed up to the target step.

Can't we keep a reference of the previous program in memory and just advance a single step?

Reply:
Indeed. That's the problem. We need to create a `VmStateIterator` in the front end. Over the weekend, I could not find out how to do that. Especially because we need all the types of that object replicated in Typescript.

Reply:
Ok, let me try. But I am not sure if we can export this. At least in the Game of Life example, the `Universe` structure only consists of `String` and `i32` types, whereas the `VmStateIterator` consists of many types that I am not allowed to export to JavaScript land.

Reply:
@Dominik1999 An approach you can use is rather than create a binding directly for `VmStateIterator`, create a handle type which carries a pointer to the `VmStateIterator` (typically done by boxing the value, but you can use references/pointers too, whatever is most appropriate). This can be represented trivially in the Wasm bindings, and then all you need to do is expose functions on that handle type to drive the iterator as normal. It's a layer of indirection, but it should be easier to generate bindings for.